src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullCoalescingAssignmentOperator.cs

```diff
@@ -109,7 +112,7 @@ BoundExpression rewriteNullCoalescingAssignmentForValueType()
                     MakeAssignmentOperator(
                         node.Syntax,
                         transformedLHS,
-                        MakeConversionNode(tmp, transformedLHS.Type, @checked: false),
+                        MakeConversionNode(tmp, transformedLHS.Type!, @checked: false),
```

Comment:
how do we know?

Reply:
Because the original left operand had a type. We can certainly add an assert after assignment above to verify though.